src/DbalStore.php

```diff
@@ -0,0 +1,303 @@
+            $resolved[$row['meta_key']] = $row['meta_key'];
+        }
+
+        $notResolvedArr = array_diff($keys, $resolved);
+        foreach ($notResolvedArr as $notResolved) {
+            $results[$notResolved] = $default;
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMultipleOrFail(array $keys)
+    {
+        KeyUtil::validateMultiple($keys);
+
+        // Normalize indices of the array
+        $keys = array_values($keys);
+
+        $data = $this->doGetMultiple($keys);
+
+        $results = array();
+        $resolved = array();
+        foreach ($data as $row) {
+            $results[$row['meta_key']] = Serializer::unserialize($row['meta_value']);
+            $resolved[] = $row['meta_key'];
+        }
+
+        $notResolvedArr = array_diff($keys, $resolved);
+
+        if (!empty($notResolvedArr)) {
+            throw NoSuchKeyException::forKeys($notResolvedArr);
+        }
+
+        return $results;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($key)
+    {
+        KeyUtil::validate($key);
+
+        try {
+            $result = $this->connection->delete($this->table, array('meta_key' => $key));
+        } catch (\Exception $e) {
+            throw WriteException::forException($e);
+        }
+
+        return $result === 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists($key)
+    {
+        KeyUtil::validate($key);
+
+        try {
+            $result = $this->connection->fetchAssoc('SELECT * FROM '.$this->table.' WHERE meta_key = ?', array($key));
+        } catch (\Exception $e) {
+            throw ReadException::forException($e);
+        }
+
+        return $result ? true : false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        try {
+            $connection = $this->connection->query('DELETE FROM '.$this->table);
+            $connection->execute();
+        } catch (\Exception $e) {
+            throw WriteException::forException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keys()
+    {
+        try {
+            $stmt = $this->connection->query('SELECT meta_key FROM '.$this->table);
+            $result = $stmt->fetchAll(\PDO::FETCH_COLUMN);
+        } catch (\Exception $e) {
+            throw ReadException::forException($e);
+        }
+
+        return $result;
+    }
+
+    public function configureTable()
+    {
+        $schema = new Schema();
+
+        $table = $schema->createTable($this->table);
+
+        $table->addColumn('id', 'integer', array('autoincrement' => true));
+        $table->addColumn('meta_key', 'string', array('length' => 255));
+        $table->addColumn('meta_value', 'object');
+        $table->setPrimaryKey(array('id'));
+        $table->addUniqueIndex(array('meta_key'));
+
+        return $table;
```

Comment:
It's used for setting up my tests, and can be used to let doctrine setup your tests, see my previous comment and https://github.com/webmozart/key-value-store/pull/15/files#diff-840e95275d34986f00b5591a212cbcd2R39